perception/lidar_transfusion/lib/preprocess/preprocess_kernel.cu

```diff
@@ -0,0 +1,258 @@
+    mask, voxels, config_.grid_y_size_, config_.grid_x_size_, config_.points_per_voxel_,
+    config_.max_voxels_, pillar_num, voxel_features, voxel_num, voxel_idxs);
+  cudaError_t err = cudaGetLastError();
+  return err;
+}
+
+__device__ int decode_field(
+  unsigned char * cloud_data, unsigned long point_offset, unsigned int field_offset,
+  unsigned char field_size, bool is_bigendian)
+{
+  int value = 0;
+  for (int i = 0; i < field_size; i++) {
+    int step = is_bigendian ? field_size - i - 1 : i;
+    value |= cloud_data[point_offset + field_offset + i] << (step * 8);
+  }
+  return value;
+}
+
+__device__ float cast_field(int & field_raw, unsigned char & field_type)
+{
+  switch (field_type) {
+    case 1:  // INT8
+      return static_cast<float>(reinterpret_cast<char &>(field_raw));
+    case 2:  // UINT8
+      return static_cast<float>(reinterpret_cast<unsigned char &>(field_raw));
+    case 3:  // INT16
+      return static_cast<float>(reinterpret_cast<short &>(field_raw));
+    case 4:  // UINT16
+      return static_cast<float>(reinterpret_cast<unsigned short &>(field_raw));
+    case 5:  // INT32
+      return static_cast<float>(reinterpret_cast<int &>(field_raw));
+    case 6:  // UINT32
+      return static_cast<float>(reinterpret_cast<unsigned int &>(field_raw));
+    case 7:  // FLOAT32
+      return reinterpret_cast<float &>(field_raw);
+    case 8:  // FLOAT64
+      return static_cast<float>(reinterpret_cast<double &>(field_raw));
+    default:
+      return 0.0;
+  }
+}
+
+__global__ void generateVoxelsInput_kernel(
+  unsigned char * cloud_data, unsigned int x_offset, unsigned int y_offset, unsigned int z_offset,
+  unsigned int intensity_offset, unsigned char x_datatype, unsigned char y_datatype,
+  unsigned char z_datatype, unsigned char intensity_datatype, unsigned char x_size,
+  unsigned char y_size, unsigned char z_size, unsigned char intensity_size, unsigned int point_step,
+  bool is_bigendian, unsigned int cloud_capacity, unsigned int points_agg, unsigned int points_size,
+  float time_lag, float * affine_past2current, float * points)
+{
+  int point_idx = blockIdx.x * blockDim.x + threadIdx.x;
+  if (point_idx >= points_size || points_agg + point_idx >= cloud_capacity) return;
+
+  unsigned long point_offset = point_idx * point_step;
+
+  int x_raw = decode_field(cloud_data, point_offset, x_offset, x_size, is_bigendian);
+  int y_raw = decode_field(cloud_data, point_offset, y_offset, y_size, is_bigendian);
+  int z_raw = decode_field(cloud_data, point_offset, z_offset, z_size, is_bigendian);
+  int intensity_raw =
+    decode_field(cloud_data, point_offset, intensity_offset, intensity_size, is_bigendian);
+
+  float x = cast_field(x_raw, x_datatype);
+  float y = cast_field(y_raw, y_datatype);
+  float z = cast_field(z_raw, z_datatype);
+  float intensity = cast_field(intensity_raw, intensity_datatype);
+
+  float x_affined = affine_past2current[0] * x + affine_past2current[1] * y +
```

Comment:
Usually (from what I have seen), the standard is `x_transformed`

Reply:
https://github.com/autowarefoundation/autoware.universe/pull/6890/commits/2eca809aa527823b66de2241382b14b45284a770